e2e/test/helpers/RetryOperationHelper.cs

```diff
@@ -78,7 +78,7 @@ public class RetryOperationHelper
                     await asyncOperation().ConfigureAwait(false);
                     return;
                 }
-                catch (Exception ex) when (!retryPolicy.ShouldRetry(counter, ex, out retryInterval))
+                catch (Exception ex) when (retryPolicy.ShouldRetry(counter, ex, out retryInterval))
```

Comment:
This doesn't seem right to me.

If we should retry, then we should not let the exception be thrown.